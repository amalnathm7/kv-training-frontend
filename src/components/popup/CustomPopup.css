.popup-overlay {
  z-index: 1;
  position: fixed;
  display: flex;
  justify-content: center;
  align-items: center;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background: rgba(0, 0, 0, .5);
  cursor: default;
}

.popup-content {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: fit-content;
  padding-left: 50px;
  padding-right: 50px;
  padding-top: 10px;
  padding-bottom: 10px;
  border: none;
  background-color: white;
}

.popup-buttons {
  width: 120%;
  display: flex;
  justify-content: space-evenly;
}