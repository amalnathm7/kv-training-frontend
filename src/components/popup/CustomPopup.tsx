import React from 'react';
import Popup from 'reactjs-popup';
import './CustomPopup.css';
import PrimaryButton from '../button/PrimaryButton/PrimaryButton';
import SecondaryButton from '../button/SecondaryButton/SecondaryButton';

type PopupPropType = {
  showPopup: boolean,
  onConfirm: () => void,
  onCancel: () => void
}

const CustomPopup: React.FC<PopupPropType> = (props) => (
  <Popup open={props.showPopup} onClose={props.onCancel}>
    <div className="popup-content">
      <h2>Are you sure?</h2>
      <p>Do you really want to delete the employee?</p>
      <div className="popup-buttons">
        <PrimaryButton onClick={props.onConfirm} type={'button'} label={'Confirm'} />
        <SecondaryButton onClick={props.onCancel} type={'button'} label={'Cancel'} />
      </div>
    </div>
  </Popup>
);

export default CustomPopup;
