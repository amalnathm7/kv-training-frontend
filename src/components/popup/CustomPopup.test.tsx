
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom';
import CustomPopup, { CustomPopupPropsType } from "./CustomPopup";

describe("CustomPopup Props Test", () => {
    test("If CustomPopup rendered correctly", () => {
        const props: CustomPopupPropsType = {
            onConfirm: () => { },
            onCancel: () => { }
        }

        const element = render(<CustomPopup {...props} />);
        expect(element).toMatchSnapshot();
    });

    test("If CustomPopup overlay click works", () => {
        const props: CustomPopupPropsType = {
            onConfirm: () => { },
            onCancel: () => { }
        }

        render(<CustomPopup {...props} />);
        const element = screen.getByTestId("primary-button-test");
        const onClick = jest.fn();
        element.onclick = onClick
        element.click();
        expect(onClick).toHaveBeenCalled();
    });
});