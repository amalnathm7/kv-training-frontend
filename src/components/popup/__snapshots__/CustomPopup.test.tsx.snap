// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomPopup Props Test If CustomPopup rendered correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="popup-overlay"
        data-testid="custom-popup-test"
      >
        <div
          class="popup-content"
        >
          <h2>
            Are you sure?
          </h2>
          <p>
            Do you really want to delete this?
          </p>
          <div
            class="popup-buttons"
          >
            <input
              class="primary-button"
              data-testid="primary-button-test"
              type="button"
              value="Confirm"
            />
            <input
              class="secondary-button"
              data-testid="secondary-button-test"
              type="button"
              value="Cancel"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="popup-overlay"
      data-testid="custom-popup-test"
    >
      <div
        class="popup-content"
      >
        <h2>
          Are you sure?
        </h2>
        <p>
          Do you really want to delete this?
        </p>
        <div
          class="popup-buttons"
        >
          <input
            class="primary-button"
            data-testid="primary-button-test"
            type="button"
            value="Confirm"
          />
          <input
            class="secondary-button"
            data-testid="secondary-button-test"
            type="button"
            value="Cancel"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
